pkg/operator/ceph/cluster/osd/spec_test.go

```diff
@@ -707,3 +708,164 @@ func getDummyDeploymentOnNode(clientset *fake.Clientset, c *Cluster, nodeName st
+	clusterInfo := &cephclient.ClusterInfo{
+		Namespace:   "ns",
+		CephVersion: cephver.Nautilus,
+	}
+	clusterInfo.SetName("testing")
+	clusterInfo.OwnerInfo = cephclient.NewMinimumOwnerInfo(t)
+	context := &clusterd.Context{Clientset: clientset, ConfigDir: "/var/lib/rook", Executor: &exectest.MockExecutor{}}
+
+	spec := cephv1.ClusterSpec{
+		Placement: cephv1.PlacementSpec{
+			"all": {
+				NodeAffinity: &v1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{
+								MatchExpressions: []v1.NodeSelectorRequirement{{
+									Key:      "role",
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{"storage-node1"},
+								}},
+							},
+						},
+					},
+				},
+			},
+			"osd": {
+				NodeAffinity: &v1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{
+								MatchExpressions: []v1.NodeSelectorRequirement{{
+									Key:      "role",
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{"storage-node1"},
+								}},
+							},
+						},
+					},
+				},
+			},
+			"prepareosd": {
+				NodeAffinity: &v1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{
+								MatchExpressions: []v1.NodeSelectorRequirement{{
+									Key:      "role",
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{"storage-node1"},
+								}},
+							},
+						},
+					},
+				},
+			},
+		},
+		Storage: cephv1.StorageScopeSpec{
+			OnlyApplySpecificOSDPlacement: false,
+		},
+	}
+
+	osdProps := osdProperties{
+		pvc: v1.PersistentVolumeClaimVolumeSource{
+			ClaimName: "pvc1",
+		},
+	}
+	osdProps.placement = cephv1.Placement{NodeAffinity: &corev1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+			NodeSelectorTerms: []corev1.NodeSelectorTerm{
+				{
+					MatchExpressions: []corev1.NodeSelectorRequirement{
+						{
+							Key:      "role",
+							Operator: v1.NodeSelectorOpIn,
+							Values:   []string{"storage-node3"},
+						},
+					},
+				},
+			},
+		},
+	},
+	}
+
+	osdProps.preparePlacement = &cephv1.Placement{NodeAffinity: &corev1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+			NodeSelectorTerms: []corev1.NodeSelectorTerm{
+				{
+					MatchExpressions: []corev1.NodeSelectorRequirement{
+						{
+							Key:      "role",
+							Operator: v1.NodeSelectorOpIn,
+							Values:   []string{"storage-node3"},
+						},
+					},
+				},
+			},
+		},
+	},
+	}
+
+	c := New(context, clusterInfo, spec, "rook/rook:myversion")
+	osd := OSDInfo{
+		ID:     0,
+		CVMode: "raw",
+	}
+
+	dataPathMap := &provisionConfig{
+		DataPathMap: opconfig.NewDatalessDaemonDataPathMap(c.clusterInfo.Namespace, "/var/lib/rook"),
+	}
+
+	// for OSD daemon
```

Comment:
lets start every comment with capital letter to be consistent with other areas in the code base.